src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -51,6 +80,39 @@ def spark_version_compatibility(spark_version: str) -> str:
     return "supported"
 
 
+class PipelinesCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "pipelines")
+        self._ws = ws
+
+    def _crawl(self) -> list[PipelineInfo]:
+        all_pipelines = list(self._ws.pipelines.list_pipelines())
+        return list(self._assess_pipelines(all_pipelines))
+
+    def _assess_pipelines(self, all_pipelines):
+        for pipeline in all_pipelines:
+            pipeline_info = PipelineInfo(pipeline.pipeline_id, pipeline.name, pipeline.creator_user_name, 1, "")
+            failures = []
+            pipeline_config = self._ws.pipelines.get(pipeline.pipeline_id).spec.configuration
+            if pipeline_config:
+                for key in pipeline_config.items():
+                    if azure_sp_conf_usage_check(str(key)):
+                        failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} pipeline.")
+                        break
```

Comment:
```suggestion
            if _azure_sp_conf_usage_check(pipeline_config):
                failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} pipeline.")
```
treat config as a dictionary